File: backend/src/main/java/com/ams/entity/Contract.java
package com.ams.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("contract")
public class Contract {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String contractNo;
    private String contractName;
    private String contractType;
    private Long vendorId;
    private Long assetId;
    private BigDecimal amount;
    private String currency;
    private LocalDate startDate;
    private LocalDate endDate;
    private String status;
    private Integer autoRenew;
    private Integer renewDays;
    private String fileUrl;
    private String remark;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/contract.ts
import http from '@/utils/http';

export interface Contract {
  id?: number;
  contractNo?: string;
  contractName: string;
  contractType: string;
  vendorId?: number;
  assetId?: number;
  amount?: number;
  currency?: string;
  startDate?: string;
  endDate?: string;
  status?: string;
  autoRenew?: number;
  renewDays?: number;
  fileUrl?: string;
  remark?: string;
}

export const getContracts = (params?: Record<string, any>) =>
  http.get<any>('/contracts', { params });

export const getExpiringContracts = (days = 30) =>
  http.get<Contract[]>('/contracts/expiring', { params: { days } });

export const getContractDetail = (id: number) =>
  http.get<Contract>(`/contracts/${id}`);

export const createContract = (data: Contract) =>
  http.post<Contract>('/contracts', data);

export const updateContract = (id: number, data: Contract) =>
  http.put<Contract>(`/contracts/${id}`, data);

export const deleteContract = (id: number) =>
  http.delete(`/contracts/${id}`);
